Kitodo/src/main/java/org/kitodo/production/helper/SearchResultGeneration.java

```diff
@@ -67,10 +69,25 @@ public HSSFWorkbook getResult() {
 
     private List<ProcessDTO> getResultsWithFilter() {
         List<ProcessDTO> processDTOS = new ArrayList<>();
+        int elasticsearchLimit = 9999;
 
         try {
-            processDTOS = ServiceManager.getProcessService().findByQuery(getQueryForFilter(ObjectType.PROCESS),
-                    ServiceManager.getProcessService().sortByTitle(SortOrder.ASC), false);
+            Long numberOfProcesses = ServiceManager.getProcessService().count(getQueryForFilter(ObjectType.PROCESS));
+            if (numberOfProcesses > elasticsearchLimit) {
+                int queriedIds = 0;
+                while (processDTOS.size() < numberOfProcesses) {
+                    RangeQueryBuilder rangeQueryBuilder = new RangeQueryBuilder(ProcessTypeField.ID.toString());
+                    rangeQueryBuilder.gte(queriedIds).lt(queriedIds+elasticsearchLimit);
+                    BoolQueryBuilder queryForFilter = getQueryForFilter(ObjectType.PROCESS);
+                    queryForFilter.should(rangeQueryBuilder);
+                    processDTOS.addAll(ServiceManager.getProcessService().findByQuery(queryForFilter,
```

Comment:
Basic consideration: If there are more than 10,000 objects, I would not first put all objects in a `Collection` and process them later, but process the objects at this point with for-each, so that Java can then free them again from memory. But, as long as you don't run out of memory, that's fine with me.

Reply:
what do you mean with "process them" ?
Putting them in the Workbook?

Reply:
yes